ionic_user_interface/src/views/ResetPassword.vue

```diff
@@ -0,0 +1,309 @@
+        if (code.charAt(i) < '0' || code.charAt(i) > '9') {
+          return false;
+        }
+      }
+      return true;
+    };
+
+    const isValidPassword = (password: string): boolean => {
+      return password.length >= 8;
+    };
+
+    const resendPasswordResetCode = throttle(async (): Promise<void> => {
+      msgBox.value?.close();
+      try {
+        const response = await axios.post(
+          process.env.VUE_APP_USER_ENDPOINT_URL + '/user/forgotPassword',
+          {
+            name: usernameText.value,
+          },
+        );
+        if (response.data.Message === 'Request password reset success') {
+          msgBoxColor.value = 'medium';
+          msgBox.value?.showMsg(`A password reset code has been sent to the user email`);
+        } else {
+          msgBox.value?.showMsg('Unable to verify: ' + response.data.Message);
+        }
+      } catch (error) {
+        if (error.response) {
+          msgBox.value?.showMsg('Error: ' + error.response.data.Message);
+        } else if (error.request) {
+          msgBox.value?.showMsg('Bad request');
+        } else {
+          msgBox.value?.showMsg('Error: ' + error.message);
+        }
+      }
+    }, 1000);
+
+    const onSubmit = throttle((event: Event): boolean => {
+      event.preventDefault();
+      msgBox.value?.close();
+      msgBoxColor.value = 'danger';
+
+      if (!isValidPasswordResetCode(passwordResetCodeText.value)) {
+        msgBox.value?.showMsg('Password reset code must be 6 digits');
+        return false;
+      }
+      if (!isValidPassword(passwordText.value)) {
+        msgBox.value?.showMsg('Password has to be at least 8 characters');
+        return false;
+      }
+      if (passwordText.value !== confirmPasswordText.value) {
+        msgBox.value?.showMsg('Passwords do not match');
+        return false;
+      }
+
+      axios
+        .post(process.env.VUE_APP_USER_ENDPOINT_URL + '/user/resetPassword', {
+          name: usernameText.value,
+          password: passwordText.value,
+          code: passwordResetCodeText.value,
+        })
+        .then((response) => {
+          if (response.data.Message === 'Password reset success') {
+            toastController
+              .create({
+                header: 'Password reset success!',
+                position: 'bottom',
+                color: 'success',
+                duration: 3000,
+                buttons: [
+                  {
+                    text: 'Close',
+                    role: 'cancel',
+                  },
+                ],
+              })
+              .then((toast) => {
+                toast.present();
+              });
+            router.push('/login');
+          } else {
+            msgBox.value?.showMsg('Unable to verify: ' + response.data.Message);
+          }
+        })
+        .catch((error) => {
+          if (error.response) {
+            if (
+              error.response.data.Message === 'ExpiredCodeException' ||
```

Comment:
Should expire code have some other action associated with it?

Reply:
From description supposed to be if the code is used already. But sometimes when its just mismatch, it returns expired too.